### PR TITLE
Update docs for Ownable2Step

### PR DESCRIPTION
**Pull request by ernestognw**
Replaces #4718

### CODE REVIEW

File: docs/modules/ROOT/pages/access-control.adoc [hunk 1]
```diff
@@ -22,6 +22,8 @@ Ownable also lets you:
 
 WARNING: Removing the owner altogether will mean that administrative tasks that are protected by `onlyOwner` will no longer be callable!
 
+Ownable is a simple and effective way to implement access control, but you should be mindful of the dangers associated with transferring the ownership to an incorrect account that can't interact with this contract anymore. An alternative to this problem is using xref:api:access.adoc#Ownable2Step[`Ownable2Step`]; a variant of Ownable that requires the new owner address to call the xref:api:access.adoc#Ownable2Step-acceptOwnership--[`acceptOwnership`] before the ownership is transferred.
```
Comment (Amxx) [inline docs/modules/ROOT/pages/access-control.adoc:25]: What do you think of phrasing it that way ?

```suggestion
Ownable is a simple and effective way to implement access control, but you should be mindful of the dangers associated with transferring the ownership to an incorrect account that can't interact with this contract anymore. An alternative to this problem is using xref:api:access.adoc#Ownable2Step[`Ownable2Step`]; a variant of Ownable that requires the new owner to explicitelly accept the ownership transfer by calling xref:api:access.adoc#Ownable2Step-acceptOwnership--[`acceptOwnership`].
```

**Review by Amxx — APPROVED** [review]
Very minor proposal, otherwise LGTM

File: docs/modules/ROOT/pages/access-control.adoc (hunk 1 above)
Comment (ernestognw) [reply docs/modules/ROOT/pages/access-control.adoc:25]: Agree! Just a minor typo:

```suggestion
Ownable is a simple and effective way to implement access control, but you should be mindful of the dangers associated with transferring the ownership to an incorrect account that can't interact with this contract anymore. An alternative to this problem is using xref:api:access.adoc#Ownable2Step[`Ownable2Step`]; a variant of Ownable that requires the new owner to explicitly accept the ownership transfer by calling xref:api:access.adoc#Ownable2Step-acceptOwnership--[`acceptOwnership`].
```

**Comment by ernestognw** [comment]
Merging since changes include only a suggestion from a previously approved review.

**Pull request merged**